src/slam_toolbox_common.cpp

```diff
@@ -63,49 +67,151 @@ void SlamToolbox::configure()
   scan_holder_ = std::make_unique<laser_utils::ScanHolder>(lasers_);
   map_saver_ = std::make_unique<map_saver::MapSaver>(shared_from_this(),
       map_name_);
+  reprocessing_transform_.setIdentity();
+
+  loadPoseGraphByParams();
+  return CallbackReturn::SUCCESS;
+}
+
+/*****************************************************************************/
+CallbackReturn SlamToolbox::on_activate(const rclcpp_lifecycle::State &)
+/*****************************************************************************/
+{
+  RCLCPP_INFO(get_logger(), "Activating");
+  sst_->on_activate();
+  sstm_->on_activate();
+  pose_pub_->on_activate();
   closure_assistant_ =
     std::make_unique<loop_closure_assistant::LoopClosureAssistant>(
     shared_from_this(), smapper_->getMapper(), scan_holder_.get(),
     state_, processor_type_);
-  reprocessing_transform_.setIdentity();
 
-  double transform_publish_period = 0.05;
-  transform_publish_period =
-    this->declare_parameter("transform_publish_period",
-      transform_publish_period);
+  double transform_publish_period = this->get_parameter("transform_publish_period").as_double();
   threads_.push_back(std::make_unique<boost::thread>(
       boost::bind(&SlamToolbox::publishTransformLoop,
       this, transform_publish_period)));
   threads_.push_back(std::make_unique<boost::thread>(
       boost::bind(&SlamToolbox::publishVisualizations, this)));
+
+  scan_filter_ =
```

Comment:
Why not use set interfaces function in the activation stage and remove the interfaces back to there?

Reply:
[612ffde29045c6170047881a9d70a3a77f5fe1d7]

- Updated to create the interfaces in the activation stage. However, `tf_` is left in configure as it it used to initialize `laser_assistant` and `pose_helper`
- Subscriber and services in slam_toolbox_localization.cpp and slam_toolbox_sync.cpp are also moved to the activation stage in the same way, since segfault occured when they were executed in inactive state.